packages/@azure/eventhubs/client/lib/linkEntity.ts

```diff
@@ -180,6 +180,28 @@ export class LinkEntity {
     }
   }
 
+  /**
+   * Removes the Sender|Receiver link and it's underlying session and also removes it from the
+   * internal map.
+   * @ignore
+   * @param {Sender | Receiver} [link] The Sender or Receiver link that needs to be removed.
+   * @returns {void} void
+   */
+  protected _removeLink(link?: Sender | Receiver): void {
+    clearTimeout(this._tokenRenewalTimer as NodeJS.Timer);
+    if (link) {
+      try {
+        //  Removing the link and it's underlying session from the internal map and releasing its listeners.
+          link.remove();
```

Comment:
This will only remove the link from the local map that the session maintains. The connection will continue to have the session in its local map.

I found a function `removeSession()` on the connection object. That would clear the session from the local map that the connection maintains. How about using that?

Reply:
Let's see if we get an answer for 
https://github.com/amqp/rhea/issues/205#issuecomment-473615598. If we do, then that would be the cleanest way to go

Reply:
I have created a PR https://github.com/amqp/rhea/pull/207 to add this function in rhea.

Reply:
>We should also call link.session.remove(). But this might not be the right place to do it. This would have been the right place if sessions and links had a 1:1 relationship, but they have 1: many relationship instead

Just FYI `link.remove();` method removes the link as well as session :)